staging/src/k8s.io/kubectl/pkg/cmd/cmd.go

```diff
@@ -173,21 +174,37 @@ func NewDefaultPluginHandler(validPrefixes []string) *DefaultPluginHandler {
 func (h *DefaultPluginHandler) Lookup(filename string) (string, bool) {
 	for _, prefix := range h.ValidPrefixes {
 		path, err := exec.LookPath(fmt.Sprintf("%s-%s", prefix, filename))
-		if err != nil || len(path) == 0 {
+		if shouldSkipOnLookPathErr(err) || len(path) == 0 {
 			continue
 		}
 		return path, true
 	}
-
 	return "", false
 }
 
+func Command(name string, arg ...string) *exec.Cmd {
+	cmd := &exec.Cmd{
+		Path: name,
+		Args: append([]string{name}, arg...),
+	}
+	if filepath.Base(name) == name {
+		lp, err := exec.LookPath(name)
+		if lp != "" && !shouldSkipOnLookPathErr(err) {
+			// Update cmd.Path even if err is non-nil.
+			// If err is ErrDot (especially on Windows), lp may include a resolved
+			// extension (like .exe or .bat) that should be preserved.
+			cmd.Path = lp
```

Comment:
Were you able to try this on a windows machine (or ask someone from sig-windows to)? Do we also need to replace `Args[0]` with the fully resolved binary? (I don't know enough about how Command works on Windows to know)

Reply:
@liggitt i am fairly sure this is fine as it is from https://github.com/golang/go/blob/14abe8aa7324bdf0e09e1dfebfb3519cc30f4918/src/os/exec/exec.go#L273-L278 :)

Reply:
ha, ok